File: Java/Shuffle the Array.java
class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public int[] shuffle(int[] nums, int n) {

        int[] ans = new int[2 * n];

        for (int i = 0; i < n; ++i) {
            ans[2 * i] = nums[i];
            ans[2 * i + 1] = nums[i + n];
        }

        return ans;
    }
}

File: Python/Shuffle the Array.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def shuffle(self, nums: List[int], n: int) -> List[int]:
        return [nums[j] for i in range(n) for j in (i, i + n)]
